src/libPersistence/BlockStorage.cpp

```diff
@@ -1648,6 +1649,24 @@ bool BlockStorage::GetMinerInfoShards(const uint64_t& dsBlockNum,
   return found;
 }
 
+bool BlockStorage::PutContractCreator(const dev::h160 address, const dev::h256 txnHash) {
+  if (!m_contractCreatorDB) {
+    return true;
+  }
+
+  lock_guard<mutex> g(m_contractCreatorMutex);
+  return m_contractCreatorDB->Insert(address.asBytes(), txnHash.asBytes());
+}
+
+dev::h256 BlockStorage::GetContractCreator(const dev::h160 address) {
+  if (!m_contractCreatorDB) {
+    return dev::h256();
+  }
+
+  lock_guard<mutex> g(m_contractCreatorMutex);
+  return dev::h256(reinterpret_cast<const unsigned char*>(m_contractCreatorDB->Lookup(address.asBytes()).c_str()), dev::h256::ConstructFromPointerType::ConstructFromPointer);
```

Comment:
could this potentially be a memory access violation if Lookup returns empty string => h256 constructor is memcpying fixed width memory?